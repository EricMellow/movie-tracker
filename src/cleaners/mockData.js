export const mockRawData = {
  "results": [
    {
      "vote_count": 3383,
      "id": 299536,
      "video": false,
      "vote_average": 8.6,
      "title": "Avengers: Infinity War",
      "popularity": 555.264536,
      "poster_path": "/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
      "original_language": "en",
      "original_title": "Avengers: Infinity War",
      "genre_ids": [
        12,
        878,
        14,
        28
      ],
      "backdrop_path": "/bOGkgRGdhrBYJSLpXaxhXVstddV.jpg",
      "adult": false,
      "overview": "Test Overview",
      "release_date": "2018-04-25"
    },
    {
      "vote_count": 513,
      "id": 427641,
      "video": false,
      "vote_average": 5.9,
      "title": "Rampage",
      "popularity": 157.877408,
      "poster_path": "/30oXQKwibh0uANGMs0Sytw3uN22.jpg",
      "original_language": "en",
      "original_title": "Rampage",
      "genre_ids": [
        28,
        12,
        878
      ],
      "backdrop_path": "/wrqUiMXttHE4UBFMhLHlN601MZh.jpg",
      "adult": false,
      "overview": "Test Overview",
      "release_date": "2018-04-12"
    }
  ]
};

export const mockCleanData = [
  {
    "id": 299536,
    "title": "Avengers: Infinity War",
    "poster": "/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
    "backdrop": "/bOGkgRGdhrBYJSLpXaxhXVstddV.jpg",
    "overview": "Test Overview",
    "rating": 8.6
  },
  {
    "id": 427641,
    "title": "Rampage",
    "poster": "/30oXQKwibh0uANGMs0Sytw3uN22.jpg",
    "backdrop": "/wrqUiMXttHE4UBFMhLHlN601MZh.jpg",
    "overview": "Test Overview",
    "rating": 5.9
  }
];
