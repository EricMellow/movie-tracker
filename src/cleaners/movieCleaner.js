const movieCleaner = (rawData) => {
  const movieData = rawData.results;
  
  const cleanData = movieData.map(movie => {
    return {
      movie_id: movie.id,
      title: movie.title,
      poster: movie.poster_path,
      backdrop: movie.backdrop_path,
      overview: movie.overview,
      vote_average: movie.vote_average,
      release: movie.release_date
    };
  });

  return cleanData;
};

export default movieCleaner;