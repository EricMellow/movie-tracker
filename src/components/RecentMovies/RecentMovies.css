.recentMovies {
  background: blue;
  height: 500px;
  display: block;
}

