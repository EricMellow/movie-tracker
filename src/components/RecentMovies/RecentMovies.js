import React, { Component } from 'react';
import './RecentMovies.css';
import { connect } from 'react-redux';
import {Card} from '../Card/Card';


export class RecentMovies extends Component {
  render() {

    const movieCards = this.props.recentMovies.map(movie => {
      return (
        <Card 
          key={movie.id}
          title={movie.title} 
          backdrop={movie.backdrop} 
          overview={movie.overview}/>
      );
    });

    return (
      <div className="recentMovies">
        RECENT MOVIES!
        {movieCards}
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  recentMovies: state.recentMovies
});

export default connect(mapStateToProps)(RecentMovies);

