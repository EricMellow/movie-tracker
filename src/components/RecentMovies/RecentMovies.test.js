import React from 'react';
import ReactDOM from 'react-dom';
import { RecentMovies } from '../RecentMovies/RecentMovies';
import { shallow } from 'enzyme';
import { mockRawData, mockCleanData } from '../../cleaners/mockData';


describe('RecentMovies', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<RecentMovies recentMovies={mockCleanData}/>, { disableLifecycleMethods: true });
  });

  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });


});