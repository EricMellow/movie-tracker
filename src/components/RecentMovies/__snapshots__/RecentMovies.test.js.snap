// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentMovies should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecentMovies
    recentMovies={
        Array [
            Object {
              "backdrop": "/bOGkgRGdhrBYJSLpXaxhXVstddV.jpg",
              "id": 299536,
              "overview": "Test Overview",
              "poster": "/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
              "title": "Avengers: Infinity War",
            },
            Object {
              "backdrop": "/wrqUiMXttHE4UBFMhLHlN601MZh.jpg",
              "id": 427641,
              "overview": "Test Overview",
              "poster": "/30oXQKwibh0uANGMs0Sytw3uN22.jpg",
              "title": "Rampage",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "RECENT MOVIES!",
        Array [
          <Unknown
            backdrop="/bOGkgRGdhrBYJSLpXaxhXVstddV.jpg"
            overview="Test Overview"
            title="Avengers: Infinity War"
/>,
          <Unknown
            backdrop="/wrqUiMXttHE4UBFMhLHlN601MZh.jpg"
            overview="Test Overview"
            title="Rampage"
/>,
        ],
      ],
      "className": "recentMovies",
    },
    "ref": null,
    "rendered": Array [
      "RECENT MOVIES!",
      Object {
        "instance": null,
        "key": "299536",
        "nodeType": "function",
        "props": Object {
          "backdrop": "/bOGkgRGdhrBYJSLpXaxhXVstddV.jpg",
          "overview": "Test Overview",
          "title": "Avengers: Infinity War",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "427641",
        "nodeType": "function",
        "props": Object {
          "backdrop": "/wrqUiMXttHE4UBFMhLHlN601MZh.jpg",
          "overview": "Test Overview",
          "title": "Rampage",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "RECENT MOVIES!",
          Array [
            <Unknown
              backdrop="/bOGkgRGdhrBYJSLpXaxhXVstddV.jpg"
              overview="Test Overview"
              title="Avengers: Infinity War"
/>,
            <Unknown
              backdrop="/wrqUiMXttHE4UBFMhLHlN601MZh.jpg"
              overview="Test Overview"
              title="Rampage"
/>,
          ],
        ],
        "className": "recentMovies",
      },
      "ref": null,
      "rendered": Array [
        "RECENT MOVIES!",
        Object {
          "instance": null,
          "key": "299536",
          "nodeType": "function",
          "props": Object {
            "backdrop": "/bOGkgRGdhrBYJSLpXaxhXVstddV.jpg",
            "overview": "Test Overview",
            "title": "Avengers: Infinity War",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "427641",
          "nodeType": "function",
          "props": Object {
            "backdrop": "/wrqUiMXttHE4UBFMhLHlN601MZh.jpg",
            "overview": "Test Overview",
            "title": "Rampage",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
